src/web/src/admin_sse.ecpp

```diff
@@ -0,0 +1,334 @@
+ \brief Open an sse connection to send any notification to UI
+*/
+#><%pre>
+#include <exception>
+#include <string>
+#include <map>
+#include <functional>
+#include <malamute.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <tntdb/connection.h>
+#include <tntdb/error.h>
+
+#include <fty_proto.h>
+#include "log.h"
+#include "utils.h"
+#include "tokens.h"
+#include "utils_web.h"
+#include "utils++.h"
+#include "str_defs.h"
+#include "dbpath.h"
+#include "db/assets.h"
+#include "cleanup.h"
+#include "helpers.h"
+
+</%pre>
+<%request scope="global">
+UserInfo user;
+bool database_ready;
+</%request>
+<%cpp>
+    // verify server is ready
+    if (!database_ready) {
+        log_debug ("Database is not ready yet.");
+        http_die ("internal-error", "Database is not ready yet, please try again after a while.");
+    }
+
+    // check user permissions
+    static const std::map <BiosProfile, std::string> PERMISSIONS = {
+            {BiosProfile::Dashboard, "R"},
+            {BiosProfile::Admin,     "R"}
+            };
+    CHECK_USER_PERMISSIONS_OR_DIE (PERMISSIONS);
+
+    // sanity checks
+    std::string checked_dc;
+    uint32_t element_id = 0;
+
+    std::string dc = qparam.param("datacenter");
+    log_debug ("Request parameters - Initial tainted values received: datacenter = '%s'\n",
+            dc.c_str ());
+
+    if (!dc.empty ())
+    {
+        if (!is_ok_name (dc.c_str ()))
+        {
+            http_die ("request-param-bad", dc.c_str ());
+        }
+    } 
+
+    checked_dc = dc;
+    std::string checked_state = "ALL";
+    std::string checked_recursive = "true";
+
+    int64_t dbid =  persist::name_to_asset_id (checked_dc);
+    if (!checked_dc.empty() && dbid == -1) {
+            http_die ("element-not-found", checked_dc.c_str ());
+    }
+    if (dbid == -2) {
+            http_die ("internal-error", "Connecting to database failed.");
+    }
+    element_id = (uint32_t) dbid;
+
+    std::map<std::string, int> desired_elements;
+    tntdb::Connection connection;
+    try {
+        connection = tntdb::connectCached (url);
+    }
+    catch (const tntdb::Error& e) {
+        log_error ("tntdb::connectCached (url = '%s') failed: %s.", url.c_str (), e.what ());
+        http_die ("internal-error", "Connecting to database failed.");
+    }
+    catch (const std::exception& e) {
+        log_error ("Exception caught: '%s'.", e.what ());
+        http_die ("internal-error", e.what ()); // TODO
+    }
+    log_debug ("Asset element id = '%" PRIu32 "'.", element_id);
+
+    if (!checked_dc.empty ()) {
+        auto asset_element = persist::select_asset_element_web_byId (connection, element_id);
+        if (asset_element.status != 1)
+            http_die ("internal-error", asset_element.msg.c_str ());
+        log_debug ("Asset element name = '%s'.", asset_element.item.name.c_str ());
+        desired_elements.emplace (std::make_pair (asset_element.item.name, 5));
+
+        if (checked_recursive.compare ("true") == 0) {
```

Comment:
checked_recursive is initialized by "true" and never re-affected.
This variable looks useless